### PR TITLE
GitHub: Run on board Unit tests in Actions

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
run tests on DUT using self hosted runner

### CODE REVIEW

**Review by ktaborowski — COMMENTED** [review]
Are there test results for this plan? I don't see any https://github.com/RobertGalatNordic/sdk-sidewalk/actions/workflows/common_run_UT_on_DUT.yml

Also some name changes regarding to https://github.com/nrfconnect/sdk-sidewalk/pull/330

File: .github/workflows/common_run_UT_on_DUT.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: Run UT on boards
+on:
+  workflow_call:
+
+jobs:
+  Run_tests_on_DUT:
```
[the remark this answers is not in the document]
Comment (RobertGalatNordic) [reply .github/workflows/common_run_UT_on_DUT.yml]: done

File: .github/workflows/common_run_UT_on_DUT.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          echo "/bin" >> $GITHUB_PATH
+          echo "/usr/share" >> $GITHUB_PATH
+          echo "/snap/bin" >> $GITHUB_PATH
+          rm -rf sidewalk artifacts twister-out* .west
+          
+      - uses: actions/checkout@v3
+        with:
+          path: sidewalk
+
+      - name: update nrf
```
[the remark this answers is not in the document]
Comment (RobertGalatNordic) [reply .github/workflows/common_run_UT_on_DUT.yml]: done

File: .github/workflows/common_run_UT_on_DUT.yml
[the remark this answers is not in the document]
Comment (RobertGalatNordic) [reply .github/workflows/common_run_UT_on_DUT.yml:1]: done

File: .github/workflows/on_PR.yml [hunk 3]
```diff
@@ -28,21 +28,24 @@ jobs:
...
           path: |
             PR_number
 
+  Build_samples_and_tests:
+    needs: ValidateCodeCompliance
+    uses: nrfconnect/sdk-sidewalk/.github/workflows/common_build_samples.yml@main
+  
+  Run_tests_on_DUT:
+    needs: Build_samples_and_tests
+    uses: nrfconnect/sdk-sidewalk/.github/workflows/common_run_UT_on_DUT.yml@main
```
[the remark this answers is not in the document]
Comment (RobertGalatNordic) [reply .github/workflows/on_PR.yml]: done

File: .github/workflows/on_commit.yml [hunk 4]
```diff
@@ -9,3 +9,7 @@ on:
 jobs:
   Build_samples_and_tests:
     uses: nrfconnect/sdk-sidewalk/.github/workflows/common_build_samples.yml@main
+
+  Run_tests_on_DUT:
+    needs: Build_samples_and_tests
+    uses: nrfconnect/sdk-sidewalk/.github/workflows/common_run_UT_on_DUT.yml@main
```
[the remark this answers is not in the document]
Comment (RobertGalatNordic) [reply .github/workflows/on_commit.yml]: done

**Review by ktaborowski — APPROVED** [review]

**Pull request merged**